custom_components/rivian/__init__.py

```diff
@@ -71,13 +71,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     vehicle_coordinators: dict[str, VehicleCoordinator] = {}
     charging_coordinators: dict[str, ChargingCoordinator] = {}
     for vin in vehicles:
-        coor = VehicleCoordinator(
-            hass=hass, client=client, vehicle_id=vehicles[vin]["id"]
-        )
+        vehicle_id = vehicles[vin]["id"]
+        coor = VehicleCoordinator(hass=hass, client=client, vehicle_id=vehicle_id)
         await coor.async_config_entry_first_refresh()
         vehicle_coordinators[vin] = coor
-        coor = ChargingCoordinator(hass=hass, client=client, vin=vin)
-        await coor.async_config_entry_first_refresh()
+        
+        coor.charging_coordinator = ChargingCoordinator(hass=hass, client=client, vehicle_id=vehicle_id)
```

Comment:
If you're creating the charging coordinator in the vehicle, you can delete this one here, otherwise we're creating it twice